File: benchmarks/executable/process_integer_exact_runtime.c
// Expected with 0 user arguments: exit 0, stdout <empty>, stderr <empty>.
// Expected with 128 user arguments: exit 1, stdout <empty>, stderr <empty>.

#include <stdint.h>

int main(int argc, char **argv) {
    (void)argv;
    const int count = argc - 1;
    return count >= 0 && count <= INT8_MAX ? 0 : 1;
}

File: benchmarks/executable/process_integer_exact_runtime.rs
// Expected with 0 user arguments: exit 0, stdout "Exact 0\n", stderr <empty>.
// Expected with 127 user arguments: exit 0, stdout "Exact 127\n", stderr <empty>.
// Expected with 128 user arguments: exit 1, stdout <empty>, stderr <empty>.

fn main() {
    let count = std::env::args_os().count().saturating_sub(1);
    let Ok(narrowed) = i8::try_from(count) else { std::process::exit(1) };
    println!("Exact {narrowed}");
}
